dear_petition/petition/tests/test_models.py

```diff
@@ -222,20 +243,75 @@ def test_refresh_record_from_data():
+    new_o_record_code = "9999"
+    updated_o_records = {
+        "Records": [
+            {
+                "Action": "CONVICTED",
+                "Description": "IMPROPER EQUIP - SPEEDOMETER",
+                "Severity": "INFRACTION",
+                "Law": "G.S. 20-123.2",
+                "Code": new_o_record_code,
+            },
+        ]
+    }
+    ciprs_record.data.get("Offense Record").update(updated_o_records)
     ciprs_record.refresh_record_from_data()
-    # assert that the file_no, sex, and offense_date field has changed on the instance
-    assert ciprs_record.file_no == file_no
-    assert ciprs_record.sex == sex
-    assert ciprs_record.offense_date.strftime(
-        "%Y-%m-%dT%H:%M:%S"
-    ) == make_datetime_aware(offense_date).strftime("%Y-%m-%dT%H:%M:%S")
+
+    raw_data_offense_records = ciprs_record.data.get("Offense Record").get("Records")
+    stored_offense_records = list(ciprs_record.offenses.first().offense_records.all())
+    assert len(stored_offense_records) == len(raw_data_offense_records)
+    assert int(new_o_record_code) == stored_offense_records[0].code
+
+    # Now lets add three offenses and call refresh
+    o_record_code_list = ["1111", "2222", "3333", "4444"]
+    o_record_int_code_list = [int(code) for code in o_record_code_list]
+    updated_o_records = {
+        "Records": [
+            {
+                "Action": "CONVICTED",
+                "Description": "IMPROPER EQUIP - SPEEDOMETER",
+                "Severity": "INFRACTION",
+                "Law": "G.S. 20-123.2",
+                "Code": o_record_code_list[0],
+            },
+            {
+                "Action": "CHARGED",
+                "Description": "SPEEDING(80 mph in a 65 mph zone)",
+                "Severity": "INFRACTION",
+                "Law": "G.S. 20-141(B)",
+                "Code": o_record_code_list[1],
+            },
+            {
+                "Action": "ARRAIGNED",
+                "Description": "SPEEDING(80 mph in a 65 mph zone)",
+                "Severity": "INFRACTION",
+                "Law": "G.S. 20-141(B)",
+                "Code": o_record_code_list[2],
+            },
+            {
+                "Action": "CONVICTED",
+                "Description": "IMPROPER EQUIP - SPEEDOMETER",
+                "Severity": "INFRACTION",
+                "Law": "G.S. 20-123.2",
+                "Code": o_record_code_list[3],
+            },
+        ]
+    }
+    ciprs_record.data.get("Offense Record").update(updated_o_records)
+    ciprs_record.refresh_record_from_data()
+
+    raw_data_offense_records = ciprs_record.data.get("Offense Record").get("Records")
+    stored_codes = []
+    stored_offense_records = list(ciprs_record.offenses.first().offense_records.all())
+    for offense_record in stored_offense_records:
+        stored_codes.append(offense_record.code)
+    assert len(stored_offense_records) == len(raw_data_offense_records)
+    assert len(stored_actions) == len(raw_data_actions)
```

Comment:
This line is supposed to be `assert len(stored_codes) == len(raw_data_codes)`, right?

Reply:
I actually think I forgot to remove that line of the code. I just want to have the assertions above 312 and below it.